### PR TITLE
nightly-core22

### PR DESCRIPTION
**Pull request by bandali**
Update base snap from core20 to core22 for Nightly

### CODE REVIEW

**Comment by seb128** [comment]
Thanks, it's not an small changeset to review it seems ok on principle. Is there a reason the debug symbols handling is being reworked as part of the more to core22 or is that just catching up with improvements made on other branches? maybe we should have a review from @lissyx before merging?

**Comment by bandali** [comment]
@seb128 thanks for the review.  The reason for changing how debug symbol handling works on `nightly` is indeed to bring it in line with the other branches (and that's basically also the majority of the changes in this PR), where we build debug symbols selectively depending on the arch.  This shouldn't really matter for Nightly builds, because we're currently only building them on amd64 on GitHub anyway, so it's more of a nice-to-have.  As I'd mentioned, this helps at least the arm64 build to pass.  I haven't yet investigated the armhf failure, but I have a suspicion it's unrelated to this and is rather due to our use of rustup on `nightly` (see the relevant comment in `stable` on our avoidance of rustup there because it causes timeouts on armhf).

The changeset is indeed large, but I put it together more or less by diffing snapcraft.yaml from `nightly` and `core22` and bringing over most of the changes from core22.  As such, I don't think a review from @lissyx is strictly necessary for this one, but if you'd feel more comfortable that way then sure.

**Review by seb128 — APPROVED** [review]
Alright, let's get it merged and iterate if needed, thanks!

**Pull request merged**
